### PR TITLE
Disable all JIT Hooks at JITServer

### PR DESCRIPTION
**Pull request by harryyu1994**
JITServer does not trigger any of the JIT hooks thus disabling them

Issue: #7508

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
@mpirvu Ready for review.

**Comment by harryyu1994** [comment]
```
#12 0x00007f252c5aa29b in masterSynchSignalHandler ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9prt29.so
#13 <signal handler called>
#14 CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> >::deallocate (this=this@entry=0x7f2528434070, pointer=0x7f2528484020, name=0x0, size=24)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/cs2/allocator.h:316
#15 0x00007f252aaf79ef in CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > >::deallocate (name=0x0, size=24, pointer=<optimized out>, 
    this=0x7f2528435490)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/cs2/allocator.h:359
#16 TR::typed_allocator<void, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > >::deallocate (n=24, p=<optimized out>, 
    this=0x7f2528435490)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/env/TypedAllocator.hpp:79
#17 TR::typed_allocator<std::_List_node<TR::ResolvedMethodSymbol*>, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > >::deallocate (n=1, 
    p=<optimized out>, this=0x7f2528435490)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/env/TypedAllocator.hpp:124
#18 std::allocator_traits<TR::typed_allocator<std::_List_node<TR::ResolvedMethodSymbol*>, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > ---Type <return> to continue, or q <return> to quit---
> >::deallocate (__n=1, __p=<optimized out>, __a=...) at /usr/include/c++/7/bits/alloc_traits.h:328
#19 std::__cxx11::_List_base<TR::ResolvedMethodSymbol*, TR::typed_allocator<TR::ResolvedMethodSymbol*, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > > >::_M_put_node (__p=<optimized out>, this=0x7f2528435490)
    at /usr/include/c++/7/bits/stl_list.h:387
#20 std::__cxx11::_List_base<TR::ResolvedMethodSymbol*, TR::typed_allocator<TR::ResolvedMethodSymbol*, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > > >::_M_clear (this=0x7f2528435490) at /usr/include/c++/7/bits/list.tcc:80
#21 std::__cxx11::_List_base<TR::ResolvedMethodSymbol*, TR::typed_allocator<TR::ResolvedMethodSymbol*, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > > >::~_List_base (this=0x7f2528435490, __in_chrg=<optimized out>)
    at /usr/include/c++/7/bits/stl_list.h:442
#22 std::__cxx11::list<TR::ResolvedMethodSymbol*, TR::typed_allocator<TR::ResolvedMethodSymbol*, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > > >::~list (this=0x7f2528435490, __in_chrg=<optimized out>)
    at /usr/include/c++/7/bits/stl_list.h:733
#23 TR::list<TR::ResolvedMethodSymbol*, CS2::shared_allocator<CS2::heap_allocator<65536ul, 12u, TRMemoryAllocator<(TR_AllocationKind)1, 12u, 28u> > > >::~list (this=0x7f2528435490, 
    __in_chrg=<optimized out>)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/infra/TRlist.hpp:32
#24 OMR::Compilation::~Compilation (this=0x7f2528434000, __in_chrg=<optimized out>)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../omr/compiler/compile/OMRCompilation.cpp:471
#25 0x00007f252a775c4c in TR::Compilation::~Compilation (this=<optimized out>, 
    __in_chrg=<optimized out>)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/compile/Compilation.hpp:66
#26 TR::CompilationInfoPerThreadBase::compile (this=this@entry=0x7f2529546590, 
    vmThread=vmThread@entry=0x1ff1d00, entry=entry@entry=0x7f242400cc70, 
    scratchSegmentProvider=...)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/CompilationThread.cpp:7754
#27 0x00007f252a8ed16d in TR::CompilationInfoPerThreadRemote::processEntry (this=0x7f2529546590, 
    entry=..., scratchSegmentProvider=...)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/JITServerCompilationThread.cpp:635
#28 0x00007f252a77195b in TR::CompilationInfoPerThread::processEntries (
    this=this@entry=0x7f2529546590)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/CompilationThread.cpp:3926
#29 0x00007f252a771e52 in TR::CompilationInfoPerThread::run (this=this@entry=0x7f2529546590)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/CompilationThread.cpp:3797
#30 0x00007f252a771efa in protectedCompilationThreadProc (compInfoPT=0x7f2529546590)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/CompilationThread.cpp:3736
#31 0x00007f252c5aaf53 in omrsig_protect ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9prt29.so
#32 0x00007f252a772354 in compilationThreadProc (entryarg=0x7f2529546590)
    at /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/vm/compiler/../compiler/control/CompilationThread.cpp:3641
#33 0x00007f252cf14202 in thread_wrapper ()
```

A testcase crashed JITServer, I guess I disabled too many things. Looking into it.

**Comment by harryyu1994** [comment]
The crash was caused by skipping this on JITServer
```
   if (!jitConfig->samplerMonitor)
       javaVM->internalVMFunctions->jvmPhaseChange(javaVM, J9VM_PHASE_NOT_STARTUP);
```
This is called inside `setUpHooks()` and we might want to keep this line on JITServer

**Comment by harryyu1994** [comment]
```
#37 <signal handler called>
#38 0x00007effcc85a078 in OMR::X86::TreeEvaluator::performCall(TR::Node*, bool, bool, TR::CodeGenerator*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#39 0x00007effcc85a282 in OMR::X86::TreeEvaluator::directCallEvaluator(TR::Node*, TR::CodeGenerator*)
    ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
---Type <return> to continue, or q <return> to quit---
#40 0x00007effcc438c6b in OMR::CodeGenerator::evaluate(TR::Node*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#41 0x00007effcc85a83c in OMR::X86::TreeEvaluator::treetopEvaluator(TR::Node*, TR::CodeGenerator*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#42 0x00007effcc438c6b in OMR::CodeGenerator::evaluate(TR::Node*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#43 0x00007effcc09854e in J9::CodeGenerator::doInstructionSelection() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#44 0x00007effcc446226 in OMR::CodeGenPhase::performInstructionSelectionPhase(TR::CodeGenerator*, TR::CodeGenPhase*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#45 0x00007effcc444b22 in OMR::CodeGenPhase::performAll() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#46 0x00007effcc4427a3 in OMR::CodeGenerator::generateCode() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#47 0x00007effcc457cd1 in OMR::Compilation::compile() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#48 0x00007effcc0d5d96 in TR::CompilationInfoPerThreadBase::compile(J9VMThread*, TR::Compilation*, TR_ResolvedMethod*, TR_J9VMBase&, TR_OptimizationPlan*, TR::SegmentAllocator const&) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#49 0x00007effcc0d6ce1 in TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary*, void*) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#50 0x00007effcdf0df53 in omrsig_protect ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9prt29.so
#51 0x00007effcc0d89d5 in TR::CompilationInfoPerThreadBase::compile(J9VMThread*, TR_MethodToBeCompiled*, J9::J9SegmentProvider&) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#52 0x00007effcc25019d in TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled&, J9::J9SegmentProvider&) ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#53 0x00007effcc0d495b in TR::CompilationInfoPerThread::processEntries() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#54 0x00007effcc0d4e52 in TR::CompilationInfoPerThread::run() ()
   from /root/Harry_Debug/openj9-openjdk-jdk8/build/linux-x86_64-normal-server-release/images/j2sdk-image/jre/lib/amd64/compressedrefs/libj9jit29.so
#55 0x00007effcc0d4efa in protectedCompilationThreadProc(J9PortLibrary*, TR::CompilationInfoPerThread*) ()
```

**Comment by harryyu1994** [comment]
@mpirvu Ready for a review.

File: runtime/compiler/runtime/codertinit.cpp [hunk 1]
```diff
@@ -451,52 +452,62 @@ void codert_init_helpers_and_targets(J9JITConfig * jitConfig, char isSMP)
...
 #endif
-   jitConfig->jitDiscardPendingCompilationsOfNatives = jitDiscardPendingCompilationsOfNatives;
-   jitConfig->jitMethodBreakpointed = jitMethodBreakpointed;
+      jitConfig->jitDiscardPendingCompilationsOfNatives = jitDiscardPendingCompilationsOfNatives;
+      jitConfig->jitMethodBreakpointed = jitMethodBreakpointed;
+      }
    jitConfig->jitIllegalFinalFieldModification = jitIllegalFinalFieldModification;
 
-   initializeCodertFunctionTable(javaVM);
+// #if defined(J9VM_OPT_JITSERVER)
```
Comment (mpirvu) [inline runtime/compiler/runtime/codertinit.cpp:485]: If these lines are not needed let's remove them

File: runtime/compiler/runtime/codertinit.cpp [hunk 2]
```diff
@@ -451,52 +452,62 @@ void codert_init_helpers_and_targets(J9JITConfig * jitConfig, char isSMP)
...
+#elif defined(TR_HOST_X86) && defined(TR_TARGET_X86) && defined(TR_TARGET_64BIT)
+   jitConfig->jitSignalHandler = jitAMD64Handler;
+#endif
+#endif
 
 #if defined(J9VM_JIT_DYNAMIC_LOOP_TRANSFER)
    jitConfig->isDLTReady = jitLookupDLT;
 #endif
-
+
```
Comment (mpirvu) [inline runtime/compiler/runtime/codertinit.cpp:510]: Extra line is not needed

File: runtime/compiler/runtime/codertinit.cpp [hunk 3]
```diff
@@ -407,7 +408,7 @@ initJitTOCForAllThreads(J9JavaVM* javaVM, UDATA jitTOC)
 
 
 void codert_init_helpers_and_targets(J9JITConfig * jitConfig, char isSMP)
-   {
+   {
```
Comment (mpirvu) [inline runtime/compiler/runtime/codertinit.cpp:411]: Extra spaces after the curly bracket that are not needed

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk11

**Comment by mpirvu** [comment]
One test on x86 had an error:
```
Testing: rc015
Test start time: 2020/04/22 13:50:57 Eastern Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdkbinary/j2sdk-image/bin/java" -XX:+UseJITServer -XX:+UseCompressedOops -Xgcpolicy:balanced -Xdebug -Xrunjdwp:transport=dt_socket,address=8888,server=y,onthrow=no.pkg.foo,launch=echo -Xjit:count=0  -Xdump    -agentlib:jvmtitest=test:rc015 -cp /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar:/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/../TKG/lib/asm-all.jar com.ibm.jvmti.tests.util.TestRunner
Time spent starting: 3174 milliseconds
Time spent executing: 7681 milliseconds
Test result: FAILED
Output from test:
 [ERR] JITServer is currently a technology preview. Its use is not yet supported
 [ERR] JVMCDRT000E Unable to locate JIT stack map - aborting VM
 [ERR] JVMCDRT001E Method: java/lang/NullPointerException.<init>(Ljava/lang/String;)V (0000000000047DE8)
 [ERR] JVMCDRT002E Failing PC: 00007F4962420223 (offset 0000000000000023), metaData = 00007F49291FF2B8
 [ERR] 17:51:50.408 0x16700j9codertvm(j9ji.110    *   ** ASSERTION FAILED ** at jswalk.c:532: ((0 ))
 [ERR] JVMDUMP039I Processing dump event "traceassert", detail "" at 2020/04/22 13:51:50 - please wait.
 [ERR] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/core.20200422.135150.22667.0001.dmp' in response to an event
 [ERR] JVMPORT030W /proc/sys/kernel/core_pattern setting "|/usr/share/apport/apport %p %s %c %d %P" specifies that the core dump is to be piped to an external program.  Attempting to rename either core or core.22679.
 [ERR] 
 [ERR] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/core.20200422.135150.22667.0001.dmp
 [ERR] JVMDUMP032I JVM requested Java dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/javacore.20200422.135150.22667.0002.txt' in response to an event
 [ERR] JVMDUMP010I Java dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/javacore.20200422.135150.22667.0002.txt
 [ERR] JVMDUMP032I JVM requested Snap dump using '/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/Snap.20200422.135150.22667.0003.trc' in response to an event
 [ERR] JVMDUMP010I Snap dump written to /home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal/openjdk-tests/TKG/test_output_15875650861791/cmdLineTester_jvmtitests_debug_openj9_none_SCC_7/Snap.20200422.135150.22667.0003.trc
 [ERR] JVMDUMP013I Processed dump event "traceassert", detail "".
>> Success condition was not found: [Return code: 0]
```

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
All tests have passed, hence merging.

**Pull request merged**